### PR TITLE
Setup end-to-end tests (Detox)

### PR DESCRIPTION
**Pull request by marek-siemieniuk-morawski**
Changes:
- Set up `detox` to work with `prod` and `beta` versions with Debug configuration (will be changed to Release later on). It was required to make some changes in the Android app source code.
- Installed all required dependencies and configured Jest to work with Typescript
- Moved some of the dependencies, like `jest`, to `devDependecies` section
- Updated README do include instruction on how to run tests
- Added `testID` properties to React elements for testing purposes

No worries, I'll squash all commits before merge :) 

For some reason, Detox doesn't work with iOS simulator. @Kamill90 is looking at it.

### CODE REVIEW

File: tests/e2e/pageObjects/pages/confirmPassword.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export class ConfirmPassword {
+    public readonly elements = {
+        input: element(by.id('confirm-transaction-passowrd')),
+        errorMessage: element(by.id('validation-error-message')).atIndex(1),
+        submit: element(by.id('submit-transaction-passowrd-confirmation'))
+    }
+
+    public async type(pin: string): Promise<void> {
+        await waitFor(this.elements.input).toBeVisible().withTimeout(10000);
```
Comment (marek-siemieniuk-morawski) [inline tests/e2e/pageObjects/pages/confirmPassword.ts:9]: the value of `withTimeout()` will be extracted to an external module later on

File: tests/e2e/specFiles/00-Onboarding.spec.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import app from '../pageObjects';
+
+describe('Onboarding', () => {
+  beforeEach(async () => {
+    // TODO: With the first execution it re-launches the app unnecessarly. Fix it.
+    await device.launchApp({ newInstance: true, delete: true });
+  });
+
+  // Due to the Detox's limitations, some of the tests are possible to execute only on iOS.
+  // For example, launching the app with specific/local language is iOS only.
```
Comment (marek-siemieniuk-morawski) [inline tests/e2e/specFiles/00-Onboarding.spec.ts:11]: This comment will be removed before merge

File: src/screens/PinFlow/ConfirmTransactionPassword.tsx [hunk 3]
```diff
@@ -84,6 +84,7 @@ class ConfirmTransactionPasswordScreen extends PureComponent<Props, State> {
         footer={
           <Button
             title={i18n._.save}
+            testID="submit-transaction-passowrd-confirmation"
```
Comment (aleparty) [inline src/screens/PinFlow/ConfirmTransactionPassword.tsx:87]: typo in password

File: src/screens/PinFlow/ConfirmTransactionPassword.tsx [hunk 4]
```diff
@@ -100,6 +101,7 @@ class ConfirmTransactionPasswordScreen extends PureComponent<Props, State> {
           </TouchableOpacity>
           <InputItem
             value={password}
+            testID="confirm-transaction-passowrd"
```
Comment (aleparty) [inline src/screens/PinFlow/ConfirmTransactionPassword.tsx:104]: typo in password

File: src/screens/PinFlow/CreateTransactionPassword.tsx [hunk 5]
```diff
@@ -72,6 +72,7 @@ export class CreateTransactionPassword extends PureComponent<Props, State> {
         footer={
           <Button
             title={i18n._.save}
+            testID="submit-create-transaction-password"
```
Comment (aleparty) [inline src/screens/PinFlow/CreateTransactionPassword.tsx:75]: typo in password

File: src/screens/PinFlow/CreateTransactionPassword.tsx [hunk 6]
```diff
@@ -88,6 +89,7 @@ export class CreateTransactionPassword extends PureComponent<Props, State> {
           </TouchableOpacity>
           <InputItem
             value={password}
+            testID="create-transaction-passowrd"
```
Comment (aleparty) [inline src/screens/PinFlow/CreateTransactionPassword.tsx:92]: typo in password

**Review by aleparty — APPROVED** [review]
lgtm

**Pull request merged**